frontend/src/components/features/terminal/terminal.tsx

```diff
@@ -1,16 +1,25 @@
 import { useSelector } from "react-redux";
 import { RootState } from "#/store";
 import { useTerminal } from "#/hooks/use-terminal";
-
 import "@xterm/xterm/css/xterm.css";
+import {
+  useWsClient,
+  WsClientProviderStatus,
+} from "#/context/ws-client-provider";
 
 interface TerminalProps {
   secrets: string[];
 }
 
 function Terminal({ secrets }: TerminalProps) {
+  const { status } = useWsClient();
   const { commands } = useSelector((state: RootState) => state.cmd);
-  const ref = useTerminal(commands, secrets);
+
+  const ref = useTerminal({
+    commands,
+    secrets,
+    disabled: status !== WsClientProviderStatus.ACTIVE,
```

Comment:
Sorry, I'm a bit confused, doesn't this disable it when the websocket isn't active? That doesn't seem to be about the runtime at all

Reply:
You may be right! My - probably silly - concern is about the opposite though: from the title of the PR/commit it sounds like it's tied to the state of the runtime, and IMHO that could disable _too much_?

Like, _during_ a session, the runtime can do some [strange stuff](https://openhands-ai.slack.com/archives/C07S376KASK/p1733028590467609), get errors, throw exceptions, or who knows, and currently it might be possible to restore it / reinit it, or even just chat with the LLM about its sad state. 😅

But the PR is not actually about the runtime? This enum can also be STOPPED or ERROR, not sure what are all cases when those get set, but as long as it's about the connection only, it's different IMHO - if you can't send, you can't send. Sorry, do I misunderstand?

Reply:
Good points. I should revise the implementation (especially for the point about different enum values). Another thing I am seeing is that the only time the state is set to ACTIVE is on init event. I don't think this would work on a refresh since it doesn't look like we're getting another init event from the server on reloads.

Reply:
Issue should be resolved. I've fixed the init event issue in #5632